src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -426,29 +426,32 @@ private synchronized EC2AbstractSlave provisionSlaveIfPossible(SlaveTemplate tem
         try {
             List<PlannedNode> r = new ArrayList<PlannedNode>();
             final SlaveTemplate t = getTemplate(label);
-
+            LOGGER.log(Level.INFO, "Attempting provision, excess workload: " + excessWorkload);
+            if (label == null) {
+                LOGGER.log(Level.WARNING, String.format("Label is null - can't caculate how many executors slave will have. Using %s number of executors", t.getNumExecutors()));
+            }
             while (excessWorkload > 0) {
-                LOGGER.log(Level.FINE, "Attempting provision, excess workload: " + excessWorkload);
-
                 final EC2AbstractSlave slave = provisionSlaveIfPossible(t);
                 // Returned null if a new node could not be created
                 if (slave == null)
                     break;
-                Hudson.getInstance().addNode(slave);
+                LOGGER.log(Level.INFO, String.format("We have now %s computers", Jenkins.getInstance().getComputers().length));
+                Jenkins.getInstance().addNode(slave);
+                LOGGER.log(Level.INFO, String.format("Added node named: %s, We have now %s computers", slave.getNodeName(), Jenkins.getInstance().getComputers().length));
                 r.add(new PlannedNode(t.getDisplayName(), Computer.threadPoolForRemoting.submit(new Callable<Node>() {
 
                     public Node call() throws Exception {
-                        try {
-                            slave.toComputer().connect(false).get();
-                        } catch (Exception e) {
-                            if (t.spotConfig != null) {
-                                LOGGER.log(Level.INFO, "Expected - Spot instance " + slave.getInstanceId()
-                                        + " failed to connect on initial provision");
-                                return slave;
+                        long startTime = System.currentTimeMillis(); // fetch starting time
+                        while (false || (System.currentTimeMillis() - startTime) < slave.launchTimeout * 1000) {
+                            try {
+                                return tryToCallSlave(slave, t);
+
+                            } catch (Exception exception) {
```

Comment:
You should not swallow the exception here. Simply don't rethrow it from the trytocallslave method.